### PR TITLE
test: expand `examples_do_not_have_gitignore` to check more files/paths

### PR DESCRIPTION
**Pull request by 0x471**
https://github.com/trailofbits/dylint/issues/1417

### CODE REVIEW

File: examples/src/lib.rs [hunk 1]
```diff
@@ -131,15 +131,39 @@ mod test {
...
+                        Some(OsStr::new(forbidden)),
+                        "Forbidden file found in examples directory: {:?}",
+                        normalized_path
+                    );
+                }
+            }
+
+            let is_allowed_directory = allowed_dirs.iter().any(|&d| normalized_path.starts_with(d));
+            if is_allowed_directory {
+            } else {
```
Comment (smoelius) [inline examples/src/lib.rs:157]: Nit:
```suggestion
            if !is_allowed_directory {
```

**Comment by smoelius** [comment]
If you can just make the lints happy, I think this PR should be good to go.

**Comment by 0x471** [comment]
Any ideas on why these two tests are failing? They passed in the previous CI run. Appreciate any insights. @smoelius

**Comment by smoelius** [comment]
Neither are related to your changes.

The `fix` test failure appears to be an instance of this known problem: https://github.com/trailofbits/dylint/issues/1353#issuecomment-2396525664

The `supply_chain` test let's me know when my dependencies' publishers change, and some of them did recently.

I'm going to rebase your branch and I expect everything to be green after that.

**Comment by 0x471** [comment]
Thanks, looks like ready to go!

**Comment by smoelius** [comment]
Thanks a lot for doing this!

**Pull request merged**